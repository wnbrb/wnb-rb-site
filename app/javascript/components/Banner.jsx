import React from 'react';
import PropTypes from 'prop-types';
import Card from 'components/Card';

import 'stylesheets/banner';

const Banner = ({ children }) => (
    <div className="banner">
        <Card className="banner-card">{children}</Card>
    </div>
);

Banner.propTypes = {
    children: PropTypes.node,
};

export default Banner;
