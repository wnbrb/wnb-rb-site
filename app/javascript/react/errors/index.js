export class UnauthorizedError extends Error {
    constructor(message) {
        super(message);
        this.name = 'UnauthorizedError';
    }
}
