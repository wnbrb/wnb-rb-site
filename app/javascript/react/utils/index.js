const MS_IN_DAY = 24 * 60 * 60 * 1000;

export const postedAtString = (createdAtString) => {
    const today = new Date();
    const postedDate = new Date(createdAtString);

    const diff = (+today - +postedDate) / MS_IN_DAY;

    if (diff <= 1) {
        return 'today';
    }

    if (diff < 7) {
        return `${Math.floor(diff)} days ago`;
    }

    if (diff < 30) {
        const weekDiff = Math.floor(diff / 7);
        const weekOrWeeks = weekDiff === 1 ? 'week' : 'weeks';

        return `${weekDiff} ${weekOrWeeks} ago`;
    }

    const monthDiff = Math.floor(diff / 30);
    const monthOrMonths = monthDiff === 1 ? 'month' : 'months';

    return `${monthDiff} ${monthOrMonths} ago`;
};
