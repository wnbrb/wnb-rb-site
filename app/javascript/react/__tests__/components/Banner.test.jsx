import React from 'react';
import { render } from '@testing-library/react';
import Banner from '../../components/Banner';

// Mock the Card component to avoid external dependencies in snapshot
jest.mock('../../components/Card', () => {
    // eslint-disable-next-line react/prop-types
    function MockCard({ children, className }) {
        return <div className={`mock-card ${className || ''}`}>{children}</div>;
    }
    return MockCard;
});

describe('Banner Component', () => {
    test('renders banner with default props', () => {
        const { container } = render(<Banner />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders banner with children content', () => {
        const { container } = render(
            <Banner>
                <h1>Welcome to our site!</h1>
                <p>This is a banner message.</p>
            </Banner>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders banner with text content', () => {
        const { container } = render(<Banner>Simple text content</Banner>);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders banner with complex nested content', () => {
        const { container } = render(
            <Banner>
                <div className="banner-content">
                    <h2>Event Title</h2>
                    <span className="date">March 15, 2024</span>
                    <button>Register Now</button>
                </div>
            </Banner>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('applies correct CSS classes', () => {
        const { container } = render(<Banner>Test content</Banner>);

        const bannerDiv = container.firstChild;
        expect(bannerDiv).toHaveClass('banner');

        const cardElement = bannerDiv.querySelector('.mock-card');
        expect(cardElement).toHaveClass('banner-card');
    });

    test('passes children correctly to Card component', () => {
        const testContent = 'Test banner content';
        const { getByText } = render(<Banner>{testContent}</Banner>);

        expect(getByText(testContent)).toBeInTheDocument();
    });
});
