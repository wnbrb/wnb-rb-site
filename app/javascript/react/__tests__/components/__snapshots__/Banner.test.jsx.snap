// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component renders banner with children content 1`] = `
<div
  class="banner"
>
  <div
    class="mock-card banner-card"
  >
    <h1>
      Welcome to our site!
    </h1>
    <p>
      This is a banner message.
    </p>
  </div>
</div>
`;

exports[`Banner Component renders banner with complex nested content 1`] = `
<div
  class="banner"
>
  <div
    class="mock-card banner-card"
  >
    <div
      class="banner-content"
    >
      <h2>
        Event Title
      </h2>
      <span
        class="date"
      >
        March 15, 2024
      </span>
      <button>
        Register Now
      </button>
    </div>
  </div>
</div>
`;

exports[`Banner Component renders banner with default props 1`] = `
<div
  class="banner"
>
  <div
    class="mock-card banner-card"
  />
</div>
`;

exports[`Banner Component renders banner with text content 1`] = `
<div
  class="banner"
>
  <div
    class="mock-card banner-card"
  >
    Simple text content
  </div>
</div>
`;
